File: Examples/FsSimpleton/Service.fs
module FsSimpleton

open System.Timers
open ServiceManager

let inline notNull value = not (obj.ReferenceEquals(value, null))

type Service() = 
    let mutable _timer : Timer = null
    let secs = 9
    let interval = float secs * 1000.; 

    member this.LogEvent e =
        ServiceContext.LogInfo("Timer fired")

    member this.StartService () = 
        _timer <- new Timer(interval)
        _timer.Elapsed.Add this.LogEvent
        _timer.Start();
        ServiceContext.LogInfo (sprintf "Timer started; will log every %A seconds" secs)

    member this.StopService () =
        if notNull _timer then
            _timer.Stop()
            _timer.Dispose()

File: Examples/DatabasePoller/Service.cs
using System.Timers;
using ServiceManager;
using ServiceManager.ServiceSupport.Configuration;

namespace DatabasePoller
{
    public class Service
    {
        Timer _timer;

        public void StartService(ServiceContext ctx)
        {
            var interval = ConfigurationManager.AppSettings["QueryInterval"];
            int parsedInterval;
            int minutes = 2;
            if (int.TryParse(interval, out parsedInterval)) {
                minutes = parsedInterval;
            }
            _timer = new Timer(minutes * 60 * 1000);
            _timer.Elapsed += (s, e) => { ServiceContext.LogInfo("Timer fired"); QueryRunner.Run();  };
            _timer.Start();
            ServiceContext.LogInfo("Timer started; will query the database every {0} minutes", minutes);
        }

        public void StopService()
        {
            if (_timer != null) {
                _timer.Stop();
                _timer.Dispose();
            }
        }
    }
}
